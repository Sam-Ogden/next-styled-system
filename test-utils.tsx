import React from "react";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import theme from "./theme";

const Theme = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options?: RenderOptions) =>
  render(ui, { wrapper: Theme, ...options });

export * from "@testing-library/react";
export { default as userEvent } from "@testing-library/user-event";
export { customRender as render };
