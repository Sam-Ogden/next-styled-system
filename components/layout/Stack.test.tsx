import { render } from "../../test-utils";
import { Stack } from ".";

describe("Stack component", () => {
  test("should render stack child margin top", () => {
    const { asFragment } = render(
      <Stack space={3}>
        <div>item1</div>
        <div>item2</div>
      </Stack>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
