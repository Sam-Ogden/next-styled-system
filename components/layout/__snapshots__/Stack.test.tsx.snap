// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack component should render stack child margin top 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 *:not(:first-child) {
  margin-top: 16px;
}

<div
    class="c0"
  >
    <div>
      item1
    </div>
    <div>
      item2
    </div>
  </div>
</DocumentFragment>
`;
